packages/flutter/lib/src/cupertino/segment_control.dart

```diff
@@ -0,0 +1,191 @@
+import 'package:flutter/widgets.dart';
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/painting.dart';
+
+class SegmentControlItem {
+  final String title;
+  final Widget content;
+
+  SegmentControlItem(this.title, this.content);
+}
+
+class SegmentControl extends StatefulWidget {
+  final List<SegmentControlItem> tabs;
+  final int activeTabIndex;
+
+  SegmentControl(this.tabs, {this.activeTabIndex = 0})
+      : assert(tabs.length > 1 && tabs.length <= 3), assert(activeTabIndex <= tabs.length - 1);
+
+  @override
+  State createState() {
+    return new _SegmentControlState();
+  }
+}
+
+class _SegmentControlState extends State<SegmentControl> {
+  int _activeTabIndex;
+
+  @override
+  void initState() {
+    super.initState();
+
+    setState(() {
+      _activeTabIndex = widget.activeTabIndex;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    Widget activeTab = widget.tabs[_activeTabIndex].content;
+
+    List<_SegmentControlItem> list = <_SegmentControlItem>[];
+
+    for (int i = 0; i < widget.tabs.length; i++) {
```

Comment:
prefer `i += 1`

Reply:
Why is that preferred?

Reply:
In some languages, the compilers aren't clever enough to optimise out the save in the postincrement operator in cases like this (i++ means "save the value of i, increment i, return the saved value", but here the saved value is ignored). So to prevent people from getting into a bad habit, we generally avoid the postincrement operator. We prefer either the pre-increment operator  or the `+=` assignment operator. Personally I think `i += 1` looks nicer and more readable than `++i` but either way is fine so long as it's not `i++`.